modules/system/controllers/Updates.php

```diff
@@ -877,6 +880,12 @@ public function onBulkAction()
             }
         }
 
+        if (in_array($bulkAction, ['disable', 'remove'])) {
+            // purge orphaned mail templates
+            MailTemplate::syncAll();
```

Comment:
Upon further review, I think what we should probably do is fire an event after a plugin is disabled or removed and then listen to that event from the System ServiceProvider to handle this syncing.

Reply:
Any suggestion on the event name @LukeTowers ? `system.plugins.pluginRemoved` ?

Should we fire a different event if it's removed (`system.plugins.pluginRemoved`) or if it's disabled (`system.plugins.pluginDisabled`) ?
`

Reply:
Or maybe just `system.plugins.afterUpdate` with the action that was done (freeze, unfreeze, disable, enable, refresh, remove) ?

Reply:
@mjauvin `system.plugins.afterDisable` & `system.plugins.afterRemove`

Reply:
@LukeTowers You don't like what I commited?

Reply:
Not really, it's way too coupled with the controller / view implementation. The events should be fired from the PluginManager inside of the appropriate methods so that they're fired even if someone is programatically removing or disabling plugins from outside of the controller context.